controllers/replicationsource.go

```diff
@@ -131,18 +131,20 @@ func (r *VolumeSnapshotBackupReconciler) setVSBRepSourceStatus(log logr.Logger)
 		if err != nil {
 			return false, err
 		}
-		reconCondition := metav1.Condition{}
-		reconConditionProgress := metav1.Condition{}
+
+		reconConditionStatus := metav1.Condition{}
+		reconConditionReason := metav1.Condition{}
+
 		for i := range repSource.Status.Conditions {
-			if repSource.Status.Conditions[i].Type == "Reconciled" {
-				reconCondition = repSource.Status.Conditions[i]
+			if repSource.Status.Conditions[i].Status == metav1.ConditionFalse {
```

Comment:
Why are we doing a `condition.Status` check here ?

Reply:
Shouldnt this be checking for condition type "Synchronizing" ?

Reply:
We are looping through but once again I think we are assuming that the first condition that you will encounter which condition status as false is the one we need, I think you need the following:
`condition.type == Synchronizing && condition.status == false`

Reply:
@shubham-pampattiwar `condition.type == Synchronizing` is used in both inProgress and completed conditions for replicationSource, so I don't think we would want to use that.

Reply:
yes thats true but fot completed state the status of the condition would be false, right ? and for sync it will be true ?